langchain4j/src/main/java/dev/langchain4j/data/document/splitter/HierarchicalDocumentSplitter.java

```diff
@@ -28,11 +28,13 @@ public abstract class HierarchicalDocumentSplitter implements DocumentSplitter {
             new DocumentBySentenceSplitter(1, 0, null, null);
 
     private static final String INDEX = "index";
+    private static final String START_INDEX = "startIndex";
 
     protected final int maxSegmentSize;
     protected final int maxOverlapSize;
     protected final Tokenizer tokenizer;
     protected final DocumentSplitter subSplitter;
+    protected final AtomicInteger startIndex = new AtomicInteger(0);
```

Comment:
What about keeping this as a local variable instead of an object field?

Reply:
Ahh, I was moving this field around during development because i thought it was getting confused. I can try putting it back inside the function.

Reply:
Have changed this - and good spot :)